From 9a77ee329d43cd69200bde74cb5e37324451e6f7 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 3 Oct 2024 22:49:56 -0400
Subject: [PATCH 07/13] Allow users to overwrite ld with environment variables

This should help in the case of cross compilation.

xref: https://github.com/conda-forge/pytorch-cpu-feedstock/pull/261
---
 tools/setup_helpers/generate_linker_script.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/setup_helpers/generate_linker_script.py b/tools/setup_helpers/generate_linker_script.py
index 11c397a9e5..e66fc19706 100644
--- a/tools/setup_helpers/generate_linker_script.py
+++ b/tools/setup_helpers/generate_linker_script.py
@@ -1,3 +1,4 @@
+import os
 import subprocess
 
 
@@ -9,8 +10,8 @@ def gen_linker_script(
         prioritized_text = [
             line.replace("\n", "") for line in prioritized_text if line != "\n"
         ]
-
-    linker_script_lines = subprocess.check_output(["ld", "-verbose"], text=True).split(
+    ld = os.environ.get("LD", "ld")
+    linker_script_lines = subprocess.check_output([ld, "-verbose"], text=True).split(
         "\n"
     )
 
